packages/aws-cdk-lib/aws-ecs/lib/credential-spec.ts

```diff
@@ -0,0 +1,139 @@
+import { IBucket } from '../../aws-s3';
+import { IParameter } from '../../aws-ssm';
+
+/**
+ * Base construct for a credential specification (CredSpec).
+ */
+export class CredentialSpec {
+  /**
+   * Get the ARN for an S3 object.
+   */
+  protected static arnForS3Object(bucket: IBucket, key: string) {
+    if (!key) {
+      throw new Error('key is undefined');
+    }
+
+    return bucket.arnForObjects(key);
+  }
+
+  /**
+   * Get the ARN for a SSM parameter.
+   */
+  protected static arnForSsmParameter(parameter: IParameter) {
+    return parameter.parameterArn;
+  }
```

Comment:
I was initially confused by this section. Can we document both clearly that they are only intended to be used by extensions of `CredentialSpec` as helper methods?

Reply:
Updated a more detailed description